sql: remove enable_row_level_security feature gate

spilchen opened the pull request:
Row-level security (RLS) was initially introduced behind a feature gate in v25.1 to enable incremental development. With the feature nearing completion—pending only #141998—this commit removes the enable_row_level_security feature gate. This change makes RLS generally available without requiring the feature to be explicitly enabled.

Informs: #137122
Epic: CRDB-11724

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143504)
<!-- Reviewable:end -->

spilchen committed:
sql: remove enable_row_level_security feature gate

Row-level security (RLS) was initially introduced behind a feature gate
in v25.1 to enable incremental development. With the feature nearing
completion—pending only #141998—this commit removes the
enable_row_level_security feature gate. This change makes RLS generally
available without requiring the feature to be explicitly enabled.

Informs: #137122
Epic: CRDB-11724

Release note: None

rafiss reviewed (approved):
nice work!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143504#-:-OMNcN67EDDtCpEazwvF:b-j4xmku)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @asg0451, @golgeek, and @srosenberg)

___
*[`pkg/cmd/roachtest/testdata/pg_regress/rowsecurity.diffs` line 1977 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/143504#-OMNc1kY74vjk7TF1rFL:-OMNc1kY74vjk7TF1rFM:b-ip9vzw) ([raw file](https://github.com/cockroachdb/cockroach/blob/cf93b5799aeca0e5acd288c2a33ed022767ce04a/pkg/cmd/roachtest/testdata/pg_regress/rowsecurity.diffs#L1977)):*
> ```
> +                  ^
> +HINT:  try \h CREATE VIEW
> +SET SESSION AUTHORIZATION regress_rls_alice;
> ```

for a followup PR: i wonder if we could work around the lack of SET SESSION AUTHORIZATION by patching the test to use SET ROLE instead. we apply other patches to the test here: https://github.com/cockroachdb/cockroach/blob/ec4bfe239ec2b7e00d67dc61a8ef84cb1725adf6/pkg/cmd/roachtest/tests/pg_regress.go#L515


<!-- Sent from Reviewable.io -->

spilchen commented:
> for a followup PR: i wonder if we could work around the lack of SET SESSION AUTHORIZATION by patching the test to use SET ROLE instead.

Thanks for the tip. I'll try that out in a follow-up PR.

TFTR!

bors r+

spilchen commented:
blathers auth-check

spilchen commented:
You shall pass
